### PR TITLE
Add notes for testing on windows

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,6 +60,13 @@ cargo run -- venv
 cargo run -- pip install requests
 ```
 
+### Testing on Windows
+
+When testing debug builds on Windows, the stack can overflow resulting in a `STATUS_STACK_OVERFLOW` error code.
+This is due to a small stack size limit on Windows that we encounter when running unoptimized builds — the release
+builds do not have this problem. We [added a `UV_STACK_SIZE` variable](https://github.com/astral-sh/uv/pull/941) to
+bypass this problem during testing. We recommend bumping the stack size from the default of 1MB to 2MB, e.g., `UV_STACK_SIZE=2000000`.
```
Comment (zanieb) [inline CONTRIBUTING.md:68]: Should we move this into a code block with proper powershell export syntax?

Comment (charliermarsh) [reply CONTRIBUTING.md:68]: Sure, makes sense.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -60,6 +60,13 @@ cargo run -- venv
...
+
 ## Running inside a docker container
```
Comment (charliermarsh) [inline CONTRIBUTING.md:70]: Can you capitalize Docker while you're here? Hah.

**Review by charliermarsh — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (zanieb) [reply CONTRIBUTING.md:70]: 😝

File: CONTRIBUTING.md (hunk 1 above)
Comment (zanieb) [reply CONTRIBUTING.md:68]: fwiw I'll need to Google this. If someone else knows it off the top of there head and gets to it before me feel free to suggest / commit it :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -60,7 +60,18 @@ cargo run -- venv
...
 
-## Running inside a docker container
+### Testing on Windows
+
+When testing debug builds on Windows, the stack can overflow resulting in a `STATUS_STACK_OVERFLOW` error code.
+This is due to a small stack size limit on Windows that we encounter when running unoptimized builds — the release
+builds do not have this problem. We [added a `UV_STACK_SIZE` variable](https://github.com/astral-sh/uv/pull/941) to
+bypass this problem during testing. We recommend bumping the stack size from the default of 1MB to 2MB, for example:
+
+```powershel
```
Comment (konstin) [inline CONTRIBUTING.md:70]:
```suggestion
```powershell
```

**Review by konstin — APPROVED** [review]

**Comment by konstin** [comment]
For anyone who comes looking at this in future: The cause is that nested futures in debug mode are massive, while they get compiled down to a reasonable size in release mode. You can debug this with some variation of (unix only):

```bash
RUSTFLAGS=-Zprint-type-sizes cargo +nightly build -p uv -j 1 > type-sizes.txt && top-type-sizes -w -s -h 10 < type-sizes.txt > sizes.txt
```

You can debug the overflowing stack with lldb using https://forums.swift.org/t/getting-stack-size-using-lldb/63207.

**Pull request merged**